java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -803,6 +803,16 @@ public final ColumnVector mergeAndSetValidity(BinaryOp mergeOp, ColumnView... co
     return new ColumnVector(bitwiseMergeAndSetValidity(getNativeView(), columnViews, mergeOp.nativeId));
   }
 
+  /**
+   * Creates a deep copy of a column while replacing the null mask. The null mask is the
+   * device_vector equivalent of the boolean column given as argument.
+   * @param validity bool column whose value is to be used as the null mask.
+   * @return Deep copy of the column with replaced null mask.
+   */    
+  public final ColumnVector copyWithValidity(ColumnView validity) {
```

Comment:
I think this functionality can be accomplished without requiring a new API.  Create a new `ColumnView` from the original with the new validityBuffer passed to the `ColumnView` constructor argument but all other parameters simply copied from the original view.  Then call `copyToColumnVector` on the result to do the deep copy to a new `ColumnVector`.  Is this new API just for convenience?

Reply:
> with the new validityBuffer passed to the ColumnView

The validity passed to the function is a `BOOL8` column with possibly null rows, and as such is not directly usable as a validity mask. 
The kernel is to convert this column down to a mask, and then set that as the validity of the resultant column.

Reply:
Ah OK, then the docs for this method need to be much more explicit about what's happening.  It's too easy to misread it as-is, or at least too easy only for me :smile:.  In addition to elaborating quite a bit in the docs as to what this is doing (and also explaining what it means when the BOOL8 column has nulls in it), maybe this would be more obvious if named something like `withBooleanColumnAsValidity`?  As it is now, it's unclear just from the name `copyWithValidity` whether the validity is coming from the other column's null mask or data interpreted as a null mask.

Reply:
> `withBooleanColumnAsValidity`

In hindsight, since we're using column handles, it isn't clear that the argument is meant to be a Boolean column. Especially from just the function signature. I'll reword the documentation, and change the method name.

Reply:
Changed to `copyWithBooleanColumnAsValidity()`. Documented null handling.